src/main/java/org/lsc/utils/ScriptingEvaluator.java

```diff
@@ -115,13 +116,33 @@ public static void contribute(String implementationName,
 
 	private ScriptableEvaluator identifyScriptingEngine(String expression) throws LscServiceException {
 		String[] parts = expression.split(":");
-		if (parts != null && parts.length > 0 && parts[0].length() < 10
-				&& instancesTypeCache.containsKey(parts[0])) {
-			return instancesTypeCache.get(parts[0]);
+		String match = matchJScriptEvaluator(parts[0]);
+		if (!match.isEmpty()) {
+			return instancesTypeCache.get(match);
 		}
 		return defaultImplementation.orElseThrow(() -> new LscServiceException("Missing Script evaluator"));
 	}
 
+	/**
+	 * Matches the prefix specifying the jscript evaluator.
+	 *
+	 * @param jscript the prefix
+	 * @return the matched jscript evaluator
+	 */
+	public String matchJScriptEvaluator(String jscript) {
+		for (String jscriptEngine : instancesTypeCache.keySet()) {
+			Pattern pattern = Pattern.compile("((\\n.*)+)" + jscriptEngine);
```

Comment:
I fear this pattern matches too many things, for example:
```
js:java.lang.System.out.println("rjs")
```
could match rjs
But also:
```
rjs:true
```
could match js

Could you add some tests to ensure your pattern matches exactly what you mean?

Reply:
Added a few unit tests for the usecases you have highlighted.